eq-author-api/middleware/upsertUser.test.js

```diff
@@ -0,0 +1,29 @@
+const upsertUser = require("./upsertUser");
+
+describe("Upsert User", () => {
+  let req, res;
+  beforeEach(() => {
+    req = { params: { questionnaireId: 1 }, auth: { sub: "new_user" } };
+    res = {
+      send: jest.fn(),
+    };
+  });
+  it("should create a user is one doesn't exist", async () => {
+    const user = await upsertUser(req, res);
+    expect(user).toMatchObject({
+      sub: "new_user",
+      id: expect.any(String),
+    });
+    expect(res.send).toHaveBeenCalledWith(200);
+  });
+  it("should update the user if any changes", async () => {
```

Comment:
We are also missing a test for when the user is already created and not changed.